### PR TITLE
Changing a person's appearance now works better

### LINKED ISSUE #1741: Genetics UI modification borked
In a recent round I was reviving a gibbed person, via the good old monkeyman braintransplant trick. Used strange reagent to revive them post surgery to skip the whole cloning thing.
Looking at the paitent, I realised they still had the name Monkey(659). 
I took the subject to genetics, and after much fiddling, was unable to change their name. Appearance and sex, yes, but not their name.

### PR DESCRIPTION
**Pull request by Crazylemon64**
Fixes #1741

:cl:Crazylemon
bugfix: Fixed a problem where the icons of a person would not correctly update when changing gender.
tweak: One's hairstyle only adjusts on gender change if it's incompatible with your new gender.
bugfix: Fixed a bug where a person's icon would only be updated if their sprite had a new layer added or removed.
bugfix: You no longer lose your name when monkeyized and reverted - you'll still look like a monkey while you're a monkey, though.
bugfix: Your organs will now match your gender when you are cloned.
bugfix: Skeletons (when a body rots) no longer retain a fleshy torso.
/:cl:

### CODE REVIEW

**Comment by Crazylemon64** [comment]
This WILL mean, that if you change someone's gender in a DNA machine, that their hair will change too, if it's not compatible. Should I adjust this?

**Comment by Yukiona** [comment]
could you also fix the all cloned vulps are male thing.

**Comment by Crazylemon64** [comment]
@Yukiona Do they show as "He" on examine, or do they just have male sprites?

**Comment by Crazylemon64** [comment]
Ah, it looks like clones always have male sprites. I'll fix that now!

**Comment by Yukiona** [comment]
Thank god, speed merge! :+1:  x 10^:+1:

**Comment by Crazylemon64** [comment]
I've done all I desire to here.

**Comment by Fox-McCloud** [comment]
>You no longer lose your name when monkeyized and reverted - you'll still look like a monkey while you're a monkey, though.

What does this mean, exactly--if you get monkeyized, will you be "monkey 234" or will you still retain your name as a monkey?

**Comment by Crazylemon64** [comment]
You'll have a monkey's name while you're a monkey, but have your normal name when you rehumanize.

**Comment by TheDZD** [comment]
And now this has merge conflicts for some reason.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -286,7 +285,7 @@ var/global/list/damage_icon_parts = list()
 	else
 		//BEGIN CACHED ICON GENERATION.
 		var/obj/item/organ/external/chest = get_organ("chest")
-		base_icon = chest.get_icon()
+		base_icon = chest.get_icon(skeleton)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/update_icons.dm]: we need to rip this out some time, now that we have a skeleton species.

Comment (Crazylemon64) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Wouldn't that mean you could make skeleton people clones with the brain_scan mode

**Pull request merged**